### ISSUE TITLE
Support for `Element::add_attribute`

### ISSUE THREAD

**Issue by grekko** [cloudflare/lol-html#205]
I'm using the [ruby](https://github.com/gjtorikian/selma) binding for `lol_html` and would really like to be able to transform `<p>Foo</p>` into `<p data-this-is-special>Foo</p>`.

I understand that the `lol_html` API currently implements [Element.html#set_attribute](https://docs.rs/lol_html/1.2.0/lol_html/html_content/struct.Element.html#method.set_attribute) with a String value, but does not yet provide a similar `Element#add_attribute`-method which writes an attribute without a String value.

Is `Element#add_attribute` something other ppl are interested in and something someone else would be willing to implement? I myself have no rust knowledge (yet) and would have to depend on someone else to provide the code here :)

**Comment by bglw** [comment]
Note that from the perspective of the HTML spec, `data-thing` and `data-thing=""` are exactly equivalent. Omitting a value parses as the empty string.

The difference is purely aesthetic, which isn't something I'm aware of existing in `lol_html` as yet since it deals with HTML semantics.

_(disclaimer: not a maintainer)_

**Comment by grekko** [comment]
> Note that from the perspective of the HTML spec, `data-thing` and `data-thing=""` are exactly equivalent.

Absolutely, thanks for clearing that up. My point is that I'll be reading the generated output and `<p data-thing>Foo</p>` is just easier and quicker to read for me personally than `<p data-thing="">Foo</p>`.
